src/rebar_erlc_compiler.erl

```diff
@@ -81,111 +89,189 @@
 %%                           'old_inets'}]}.
 %%
 
--spec compile(rebar_app_info:t()) -> 'ok'.
-compile(AppInfo) ->
+%% @equiv compile(AppInfo, []).
+
+-spec compile(rebar_app_info:t()) -> ok.
+compile(AppInfo) when is_tuple(AppInfo), element(1, AppInfo) == app_info_t ->
```

Comment:
i'll drop the `is_tuple/1` but currently the state_t and app_info_t records aren't exported